execution/src/execution.rs

```diff
@@ -345,6 +346,102 @@ impl<R: ExecutionRpc> ExecutionClient<R> {
         }
         Ok(logs)
     }
+
+    pub async fn get_fee_history(
+        &self,
+        block_count: u64,
+        last_block: u64,
+        reward_percentiles: &[f64],
+        payloads: &BTreeMap<u64, ExecutionPayload>,
+    ) -> Result<Option<FeeHistory>> {
+        let helios_latest_block = *payloads.last_key_value().unwrap().0;
+        let helios_oldest_block = *payloads.first_key_value().unwrap().0;
```

Comment:
Unwraps will cause the program to panic (abort in release mode) - we should gracefully return an error here instead.

Example:

```suggestion
        let helios_latest_block = *payloads.last_key_value();
        let helios_latest_block = helios_latest_block.ok_or(BlockNotFoundError::new(BlockTag::Latest))?.0;
        let helios_oldest_block = *payloads.first_key_value();
        let helios_oldest_block = helios_oldest_block.ok_or(BlockNotFoundError::new(BlockTag::Number(0)))?.0;
```

(I would also allow `BlockNotFoundError` to accept an optional `BlockTag` so we can construct an error without knowing the actual block)